### PR TITLE
Improve Error Messages for all commands

### LINKED ISSUE #159: [PE-D] Error message for index 0 is inconsistent with other invalid index
Might want to standardise as both are invalid index technically, `doneTask 0` is not exactly an invalid command format.

-------------
Labels: `severity.VeryLow` `type.FunctionalityBug`
original: alissayarmantho/ped#7

### PR DESCRIPTION
**Pull request by Superbestron**
For commands involving index:
1. If user enters index > highest index in current list, gives "The task provided is invalid".
2. If user enters index <= 0, gives "Index entered must be greater than zero".
3. If user enters index with 10 or more characters, gives "Index entered must be less than 10 characters". (guard against overflow)
4. If user enters other non-numbers/does not provide index, gives "Invalid Command Format ...."
Fixes #159 #175 #176 #177 #178 #179 #180 #181 #182

### CODE REVIEW

File: src/main/java/nurseybook/logic/commands/EditCommand.java [hunk 1]
```diff
@@ -94,7 +94,7 @@ public CommandResult execute(Model model) throws CommandException {
         Elderly elderlyToEdit = lastShownList.get(index.getZeroBased());
         Elderly editedElderly = createEditedElderly(elderlyToEdit, editElderlyDescriptor);
 
-        if (elderlyToEdit.isSameElderly(editedElderly)) {
+        if (elderlyToEdit.equals(editedElderly)) {
```
Comment (CraveToCode) [inline src/main/java/nurseybook/logic/commands/EditCommand.java:97]: Could I check what is the rationale for this change?

File: src/main/java/nurseybook/logic/parser/ParserUtil.java [hunk 2]
```diff
@@ -33,18 +33,34 @@
...
     public static Index parseIndex(String oneBasedIndex) throws ParseException {
         String trimmedIndex = oneBasedIndex.trim();
+
+        if (!trimmedIndex.matches("[0-9,-]+")) {
+            throw new ParseException(MESSAGE_UNKNOWN_INDEX);
+        }
+        // number with 10 digits might overflow
+        if (trimmedIndex.length() > 9) {
+            throw new ParseException(MESSAGE_INDEX_TOO_EXTREME);
+        }
```
Comment (CraveToCode) [inline src/main/java/nurseybook/logic/parser/ParserUtil.java:55]: Perhaps for integers that are 10 digits long but less than max_int, we could parse them as a `long` and check if there is overflow when we convert it back to an `int`. That way we do not have an incorrect error message for 10 digit long integers that are still below max_int.

**Review by CraveToCode — CHANGES_REQUESTED** [review]
LGTM! Just a few suggestions to improve handling coverage of all integers.

File: src/main/java/nurseybook/logic/commands/EditCommand.java (hunk 1 above)
Comment (Superbestron) [reply src/main/java/nurseybook/logic/commands/EditCommand.java:97]: It's to improve error handling, if user enters the same field during editElderly, it will prompt an error saying "Fields entered are the same as before".

**Review by CraveToCode — APPROVED** [review]
